fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/api/client/gametest/v1/TestScreenshotComparisonOptions.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.gametest.v1;
+
+import com.google.common.base.Preconditions;
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.client.texture.NativeImage;
+
+import net.fabricmc.fabric.impl.client.gametest.TestScreenshotComparisonOptionsImpl;
+
+/**
+ * Options for comparing screenshots.
+ *
+ * <p>By default, screenshots are compared with a fuzzy comparison which allows screenshots to be 0.5% different from
+ * the template image. See {@link TestScreenshotComparisonAlgorithm#defaultAlgorithm()} for details on how this is
+ * implemented. To change the threshold for how different the screenshot can be, set the algorithm to
+ * {@link TestScreenshotComparisonAlgorithm#meanSquaredDifference(float)}. To force the screenshot to be exactly the
+ * same as your template, use {@link TestScreenshotComparisonAlgorithm#exact()}. It is recommended to stick with fuzzy
+ * matching unless you specifically require an exact match, because the exact screenshot can vary slightly on different
+ * frames and in rare cases other factors such as the GPU/driver version.
+ *
+ * <p>Template images must be fully opaque, this API will throw if the template image has any transparent pixels.
+ *
+ * <p>Template images loaded from a path are expected to be in the {@code templates} directory inside the resources
+ * directory of the mod which registered the gametest. If the template image is not found, and the
+ * {@code fabric.client.gametest.testModResourcesPath} system property is set to the resources folder of the mod which
+ * registered the gametest, then the screenshot will be saved to the {@code templates} folder so that it can be used
+ * next time the gametest is run.
```

Comment:
I'm wondering if loom could set this system property by default?

Reply:
Yes, we can assume the default path and allow the user to override it in the DSL, can you open an loom issue so its not forgotten about?